### ISSUE TITLE
Fail to open config.pb.txt when running yolo_camera_detector_test

### ISSUE THREAD

**Issue by XiranBai** [ApolloAuto/apollo#5706]
System information

    **OS Platform and Distribution **: Host: Linux Ubuntu 16.04/Docker: 14.04
    **Apollo installed from **: docker
    **Apollo version **: 3.0

I am trying to run the yolo_camera_detector_test in module/perception/obstacle/camera/detector/yolo_camera_detector after building it with Bazel to see my own image result, but I got the following error:

~~~
_[NVBLAS] NVBLAS_CONFIG_FILE environment variable is NOT set : relying on default config filename 'nvblas.conf'
[NVBLAS] Cannot open default config file 'nvblas.conf'
[NVBLAS] Config parsed
[NVBLAS] CPU Blas library need to be provided
Running main() from gmock_main.cc
[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from YoloCameraDetectorTest
[ RUN      ] YoloCameraDetectorTest.model_init_test
WARNING: Logging before InitGoogleLogging() is written to STDERR
E0907 15:41:44.330113  5446 file.h:97] Failed to open file modules/perception/model/camera/yolo_camera_detector_config.pb.txt in text mode.
E0907 15:41:44.330178  5446 file.h:140] Failed to open file modules/perception/model/camera/yolo_camera_detector_config.pb.txt in binary mode.
F0907 15:41:44.330260  5446 yolo_camera_detector.cc:41] Check failed: GetProtoFromFile(FLAGS_yolo_camera_detector_config, &config_) 
*** Check failure stack trace: ***
Aborted (core dumped)_
~~~

It looks like there is problem with the **yolo_camera_detector_config.pb.txt**, but I checked that file and it 
seems normal. Could somebody help me with this issue? Or if anyone know a better way to run the perception module with my own data, please share with me. Thanks!

**Comment by muleisheng** [comment]
You could set the path of nvblas.conf by using _export NVBLAS_CONFIG_FILE=/usr/local/cuda_.
You could not find the file in /usr/local/cuda. You may need to create one first.

**Comment by XiranBai** [comment]
Hi muleisheng, thanks for your reply. I created the nvblas.conf, now the first 3 lines of warning became:
~~~
_[NVBLAS] NVBLAS_CONFIG_FILE environment variable is set to '/usr/local/cuda'
[NVBLAS] Config parsed
[NVBLAS] CPU Blas library need to be provided_
~~~

I put
~~~ 
NVBLAS_CPU_BLAS_LIB  /usr/lib/libopenblas.so 
~~~
in the nvblas.conf, but it still says 'CPU Blas library need to be provided', and the rest of the error messages are still the same. Could you share with me if you have any idea what is going on? Thank you very much!

**Comment by XiranBai** [comment]
I am closing this as the rest of the problem seems to because the file path is wrong

**Comment by xinwf** [comment]
@sopXx, hey, how do you fix this problem, I also meet the same problem as you, can you share your method here?

**Comment by HouYuu** [comment]
@xinwf export NVBLAS_CONFIG_FILE=/usr/local/cuda

**Comment by xinwf** [comment]
@HouYuu, I have done this, it seems that works right, but another problem `CPU Blas library need to be provided` still exists, how to solve it?

**Comment by HouYuu** [comment]
I have never seen the problem like that，you should post your detailed  operation steps and system version information




| |
houyu
Samsung Electronics  - 
Software Engineer
18189118087
邮箱：houyu011@163.com
|

Signature is customized by Netease Mail Master

On 11/21/2018 13:53, 辛文飞 wrote:

@HouYuu, I have done this, it seems that works right, but another problem CPU Blas library need to be provided still exists, how to solve it?

**Comment by xinwf** [comment]
detailed info is at this issue, #6013 
if I set the `NVBLAS_CONFIG_FILE=/usr/local/cuda`，then the log is config is parsed, but the CPU BLAS library problem still exists, if I set the set the `NVBLAS_CONFIG_FILE=/usr/local/cuda/nvblas.conf`, it won't generate such log, no tips about the CPU, it looks like everything is ok, but when I search this problem in issues, they all answer this question with the reply setting the `NVBLAS_CONFIG_FILE=/usr/local/cuda`, so, I set it with this common setting, keeping consistent with all of you, I don't konw which is right.

**Comment by HouYuu** [comment]
@xinwf Have you tried to use the released version of Apollo 2.5 or 3.0?

**Comment by xinwf** [comment]
@HouYuu, no, I always use develop version from 2.5 to 3.0
